src/main/java/book/store/repository/shoppingcart/ShoppingCartRepository.java

```diff
@@ -0,0 +1,11 @@
+package book.store.repository.shoppingcart;
+
+import book.store.model.ShoppingCart;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface ShoppingCartRepository extends JpaRepository<ShoppingCart, Long> {
+    @Query("SELECT sc FROM ShoppingCart sc WHERE sc.user.id = :id")
+    ShoppingCart getUserById(@Param("id") Long userId);
```

Comment:
Why your method called getUserById if you select shopping cart by user id? 
findByUserId - is better